airflow/models/dagrun.py

```diff
@@ -1538,19 +1538,11 @@ def schedule_tis(
                 and not ti.task.outlets
             ):
                 dummy_ti_ids.append((ti.task_id, ti.map_index))
-            elif (
-                ti.task.start_trigger is not None
-                and ti.task.next_method is not None
-                and not ti.task.on_execute_callback
-                and not ti.task.on_success_callback
-                and not ti.task.outlets
-            ):
+            elif ti.task.start_from_trigger is True and ti.task.start_trigger_args is not None:
+                ti.start_date = timezone.utcnow()
                 if ti.state != TaskInstanceState.UP_FOR_RESCHEDULE:
                     ti.try_number += 1
-                ti.defer_task(
-                    exception=TaskDeferred(trigger=ti.task.start_trigger, method_name=ti.task.next_method),
-                    session=session,
-                )
+                ti.defer_task_from_scheduler(session=session, start_trigger_args=ti.task.start_trigger_args)
```

Comment:
This is a bit weird anyways. `defer_task_from_scheduler` (or the equivalent method on TI) already has `self.task.start_trigger_args` - we don't need to pass it.

Reply:
ah yes, that's indeed weird 🤔

Reply:
Just confirmed this can be removed. Thanks!